pymc/model/core.py

```diff
@@ -1387,11 +1407,17 @@ def create_value_var(
 
         # Make the value variable a transformed value variable,
         # if there's an applicable transform
-        if transform is UNSET:
+
+        if default_transform is UNSET:
             if rv_var.owner is None:
-                transform = None
+                default_transform = None
             else:
-                transform = _default_transform(rv_var.owner.op, rv_var)
+                default_transform = _default_transform(rv_var.owner.op, rv_var)
+
+        if transform is UNSET:
```

Comment:
We should add a transition branch, in the case the user set `transform=None` but left `default_transform=UNSET`, which is the code someone would write if they wanted to disable the default transforms in the old API.

If they do that we can, for now, set `default_transform=None`, and issue a UserWarning about the new behavior, explaining that if they want to disable default transforming they will have to set `default_transform=None` in the future instead.